packages/react-aria-components/src/Switch.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaSwitchProps, mergeProps, useFocusRing, useHover, usePress, useSwitch, VisuallyHidden} from 'react-aria';
+import {ContextValue, RenderProps, SlotProps, useContextProps, useRenderProps} from './utils';
+import React, {createContext, ForwardedRef, forwardRef, useState} from 'react';
+import {useToggleState} from 'react-stately';
+
+export interface SwitchProps extends Omit<AriaSwitchProps, 'children'>, RenderProps<SwitchRenderProps>, SlotProps {}
+
+export interface SwitchRenderProps {
```

Comment:
What criteria are we using for which components handle tracking hover/focus/selected/press states internally and which one won't? I know some of it comes from our aria hooks returning things like `isSelected`/`isPressed` but here we expose/handle the hover state for Switch but didn't do so for TextField (more specifically Input)

Reply:
It's a good question. We should probably do an audit and verify that all the states are added where appropriate. I was a bit hesitant to add hover everywhere because of potential performance impact even when you don't use it, but consistency is probably more important I guess.

Reply:
> I was a bit hesitant to add hover everywhere because of potential performance impact even when you don't use it, but consistency is probably more important I guess

I agree with consistency being more important, I feel that people would be confused as to why some component don't  support the interaction state data selectors. The potential performance impact is def important though, we could do some testing around that. 

As for what would be responsible for adding hover, I'm leaning towards having the actual component (aka `TextField`/`ComboBox`) handle that since `Input` itself probably won't be used standalone outside of an aria component I'm guessing

Reply:
Yeah I think it's just defining what the patterns are and where to draw the line. For example, technically you could have a hover state on a Popover or ProgressBar, but is this likely? Probably not. So do we draw the line at "interactive" components? Anything that's pressable? What about something like an individual segment within a DateField? Maybe that's splitting hairs too finely but would be good to develop a good idea of where we want these states and where it is overkill.